gym-unity/gym_unity/envs/__init__.py

```diff
@@ -364,9 +364,8 @@ def _check_agents(self, n_agents: int) -> None:
 
     def _sanitize_info(self, step_result: BatchedStepResult) -> BatchedStepResult:
         n_extra_agents = step_result.n_agents() - self._n_agents
-        if n_extra_agents < 0 or n_extra_agents > self._n_agents:
+        if n_extra_agents < 0:
```

Comment:
I'm not 100% it's safe to remove this, but it's what was triggering in the single agent case (with 2 "done" agents)